source/developers-guide/plugin-extension-by-plugin/index.md

```diff
@@ -0,0 +1,748 @@
+---
+layout: default
+title: Enhance a plugin with a plugin
+github_link: developers-guide/plugin-extendion-by-plugin/index.md
+indexed: true
+tags:
+  - extend
+  - plugin
+  - custom
+  - products
+---
+
+<div class="toc-list"></div>
+
+## Introduction
+This document will describe how existing plugins can be extended.
+
+With the objective of implement our own option type in CustomProducts which we can use in a Template and make it possible for a customer to upload files with special mime types.
+
+To achieve the objective we decorate a service to extend the function, extend ExtJs by Smarty blocks and implement new ExtJs and Smarty templates.
+
+
+In this wiki article we expand the plugin Custom Product (v2) with the following functions on base of the 5.2 plugin system: 
+
+ - Decoration of the FileTypeWhiteList.php to add other file extensions to the whitelist. 
+ - Adding a custom option type. 
+  
+
+#### Skills: 
+- 5.2 plugin system 
+- PHP
+- ExtJs
+- Template engine Smarty
+
+## The file and directory structure
+![Directory and file overview](img/fileOverview.png)
+
+## Create the plugin basics
+At first create the base structure of the plugin, including plugin.xml, config.xml, services.xml and the SwagExtendCustomProducts.php.
+This files are the base of each plugin. 
+
+For more information about <a href="{{ site.url }}/developers-guide/plugin-system/">the 5.2 Plugin system</a>
+
+### SwagExtendCustomProducts/plugin.xml 
+contains the base information about the plugin, the label in englisch and german, the plugin version, the required shopware version and the changelog. 
+This file is equal to the old plugin.json.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<plugin xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/shopware/5.2/engine/Shopware/Components/Plugin/schema/plugin.xsd">
+    <label lang="de">Erweiterung für CustomProducts (v2)</label>
+    <label lang="en">Extension for CustomProducts (v2)</label>
+    <version>1.0.0</version>
+    <copyright>(c) by shopware AG</copyright>
+    <license>proprietary</license>
+    <link>http://store.shopware.com</link>
+    <author>shopware AG</author>
+    <compatibility minVersion="5.2.0"/>
+    <changelog version="1.0.0">
+        <changes lang="de">Erstveröffentlichung;</changes>
+        <changes lang="en">First release;</changes>
+    </changelog>
+</plugin>
+```
+
+### SwagExtendCustomProducts/Resources/config.xml
+describes possible global plugin settings.
+
+In this file you can easy implement a bunch of plugin setting which the customer can use to control the pluign.
+Each element node represent a single plugin setting. 
+Also you can translate the label and the description in your language.
+
+Possible types for elements are:
```

Comment:
don't list the options here. we have an own article for the configuration. just link it here